### PR TITLE
Add missing support for source argument in setSourceDataAtCell

### PR DESCRIPTION
**Pull request by adrianszymanski89**
### Context

I added missing support for source argument in `setSourceDataAtCell` if changes are an array, similar as it is done in `setDataAtCell `.

### How has this been tested?

By running tests locally.

### Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. https://github.com/handsontable/handsontable/issues/8599

### Affected project(s):
- [X] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:
- [X] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

[skip changelog]

### CODE REVIEW

File: handsontable/test/types/methods.types.ts [hunk 1]
```diff
@@ -135,7 +135,7 @@ hot.setDataAtRowProp([[123, 'foo', 'foo'], [123, 'foo', 'foo']], 'foo');
 hot.setDataAtRowProp(123, 'foo', 'foo', 'foo');
 hot.setSourceDataAtCell([[1, 'foo', 'foo']]);
 hot.setSourceDataAtCell(123, 123, 'foo', 'sourceString');
-hot.setSourceDataAtCell(123, 123, 'foo');
+hot.setSourceDataAtCell(123, 123, 'foo', 'sourceString');
```
Comment (aninde) [inline handsontable/test/types/methods.types.ts:138]: Why is this change here, since it copies the above line 137? Source argument is optional, so the test, where it is not used should remain.

**Review by aninde — CHANGES_REQUESTED** [review]

File: handsontable/test/types/methods.types.ts (hunk 1 above)
Comment (adrianszymanski89) [reply handsontable/test/types/methods.types.ts:139]: Right, I edited the wrong line. I also forgot to add the test variation for an array of changes. Now tests should be complete.

**Review by wszymanski — APPROVED** [review]

**Review by aninde — APPROVED** [review]
Thank you for for detailed test cases 👍
